guppylang/prelude/builtins.py

```diff
@@ -69,25 +80,27 @@ class array(Generic[_T, _n]):
 
 @guppy.extend_type(builtins, bool_type_def)
 class Bool:
-    @guppy.hugr_op(builtins, logic_op("And", [tys.TypeArg(tys.BoundedNatArg(n=2))]))
+    @guppy.hugr_op(builtins, logic_op("And", 2))
     def __and__(self: bool, other: bool) -> bool: ...
 
     @guppy.custom(builtins, NoopCompiler())
     def __bool__(self: bool) -> bool: ...
 
-    @guppy.hugr_op(builtins, int_op("ifrombool"))
+    @guppy.hugr_op(builtins, int_op("ifrombool", [bool], [int], n_vars=0))
```

Comment:
If you go with my signature suggestion from above, you wouldn't need to duplicate the signature in the decorator.

Also, I'm not a big fan of using Python types here...

Reply:
Done, we still need some extra parameters to specify the required arguments (since those don't correspond to guppy op args), but the input/output rows are gone.